test/release/examples/primers/specialMethods.chpl

```diff
@@ -0,0 +1,172 @@
+
+record ExampleRecord2 {
+  var exampleField: int;
+}
+proc ExampleRecord2.secondaryMethod() { }
+
+// This primer will use the secondary method form, but all of the special
+// methods can also be written as primary methods.
+
+// First we will declare a simple record with a field that is a tuple of
+// integers.  We'll add special methods and iterators to this record later.
+record R {
+  param size: int = 10;
+  var vals: size*int;
+}
+
+/*
+  Initializers and Deinitializers
+  -------------------------------
+*/
+
+// An initializer is called when creating an instance of the class or record,
+// for example with the ``new`` keyword. An initializer accepting zero
+// arguments is called a *default initializer*.  An initializer that takes
+// another instance of the same class or record is called a *copy initializer*.
+
+// The ``deinit`` method will deinitialize a record when it leaves scope,
+// or a class when ``delete`` is called on it. If the class or record
+// contained any unmanaged classes, open files, etc. this method would be
+// the place to delete them or otherwise clean them up. See the :doc:`records`
+// primer for more details on initializers and deinitializers.
+
+/*
+  The ``this`` Accessor
+  ---------------------
+*/
+
+// The ``this`` method gives the record the ability to be accessed like an
+// array.  Here we use the index to choose a tuple element.
+proc R.this(n: int) ref {
+  if n < 1 || n > size then
+    halt("index out of bounds accessing R");
+  return vals[n];
+}
+
+var r = new R();
+
+// All functions and methods in Chapel can be called using either parenthesis
+// or square brackets. Here we use square brackets to style it as an array
+// access.
+r[1] = 1;
+r[3] = 3;
+
+writeln(r.vals);
+
+/*
+  Default Iterators
+  -----------------
+*/
+
+// An iterator named ``these`` is automatically called when a record or class
+// instance is used in the iterator position of a ``for`` loop.
+iter R.these() ref {
+  for i in 1..size {
+    yield vals[i];
+  }
+}
+
+for val in r {
+  val += 1;
+}
+writeln(r.vals);
+
+// Classes and records can also define parallel iterators including
+// leader/follower iterator pairs and standalone parallel iterators. For
+// more information on parallel iterators, see the :doc:`parIters` primer.
+
+/*
+  IO Methods
+  ----------
+*/
+
+// Define how to write an instance of R to a channel. We'll write the
+// 'vals' tuple between asterisks. See section
+// :ref:`readThis-writeThis-readWriteThis` for more information  on the
+// ``writeThis``, ``readThis``, and ``readWriteThis`` methods.
+proc R.writeThis(ch: channel) {
+  ch.write("*", vals, "*");
+}
+
+{
+  // Open the file in a new block so that deinitializers
+  // will close it at the end of the block
+  var f = open("tempfile.txt", iomode.cw);
+  var ch = f.writer();
+  ch.writeln(r);
+}
+
+// Define how to read an instance of R from a channel. We'll read the
```

Comment:
Ditto.  It almost feels as though the names of the routines you're describing should be mentioned as the subject of these sentences?  But I'm not sure I'm parsing them correctly.  Or maybe they're just written as one would comment code that wasn't going to be rendered online as a primer?